File: GC.Domain/Gardens/Sectors/Credits/SectorCredit.cs
using System;

namespace GC.Domain.Gardens.Sectors.Credits
{
    public class SectorCredit
    {
        public Guid Id { get; }
        public Gardener Gardener { get; }
        public GardenSector Sector { get; }
        public Int32 Credit { get; }

        public SectorCredit(Guid id, Gardener gardener, GardenSector sector, Int32 credit)
        {
            Id = id;
            Gardener = gardener;
            Sector = sector;
            Credit = credit;
        }
    }
}

File: GC.WebSpace/ClientApp/domain/gardens/sectorCredits/sectorCredit.ts
import { Gardener } from "../gardeners/gardener";
import { GardenSector } from "../gardenSectors/gardenSector";

export class SectorCredit {
    constructor(
        public readonly id: string,
        public readonly gardener: Gardener,
        public readonly sector: GardenSector,
        public readonly credit: number
    ) { }
}

export namespace SectorCredit {
    export function toModel(data: any): SectorCredit {
        return new SectorCredit(data.id, Gardener.toModel(data.gardener), GardenSector.toModel(data.sector), Number(data.credit));
    }

    export function toModels(data: any[]): SectorCredit[] {
        return data.map(toModel);
    }
}